.github/workflows/deploy-to-eks.yaml

```diff
@@ -58,13 +75,25 @@ jobs:
       - name: Setup juju
         run: |
           juju add-k8s kubeflow --client
-          juju bootstrap --no-gui kubeflow kubeflow-controller
+          juju bootstrap kubeflow kubeflow-controller
           juju add-model kubeflow
       
       - name: Test bundle deployment
         run: |
-          tox -vve test_bundle_deployment-1.7 -- --model kubeflow --keep-models -vv -s
-      
+          tox -vve test_bundle_deployment-1.8 -- --model kubeflow --keep-models -vv -s
+
+      - name: Run Kubeflow UATs
+        run: |
+          git clone https://github.com/canonical/charmed-kubeflow-uats.git ~/charmed-kubeflow-uats
+          cd ~/charmed-kubeflow-uats
+          git checkout ${{ env.UATS_BRANCH }}
+          tox -e kubeflow-remote
+
+      # On failure, capture debugging resources
```

Comment:
I know it's from AKS (:smiling_face_with_tear: ) but should we either fix the message or fix the `if:` statement since ATM the message state "on failure" but the step run `Always`. On top of that, should we update all the debugging steps like this
```diff
-         if: failure()
+         if: failure() || cancelled()
```
because there are cases where the run can get stuck and you may want to cancel, so you can get the logs. WDYT about it?